apps/web/utils/outlook/client.ts

```diff
@@ -0,0 +1,164 @@
+import { Client } from "@microsoft/microsoft-graph-client";
+import type { User, Photo } from "@microsoft/microsoft-graph-types";
+import { saveTokens } from "@/utils/auth";
+import { env } from "@/env";
+import { createScopedLogger } from "@/utils/logger";
+import { SCOPES } from "@/utils/outlook/scopes";
+import { SafeError } from "@/utils/error";
+
+const logger = createScopedLogger("outlook/client");
+
+type AuthOptions = {
+  accessToken?: string | null;
+  refreshToken?: string | null;
+  expiryDate?: number | null;
+  expiresAt?: number | null;
+};
+
+// Wrapper class to hold both the Microsoft Graph client and its access token
+export class OutlookClient {
+  private client: Client;
+  private accessToken: string;
+
+  constructor(accessToken: string) {
+    this.accessToken = accessToken;
```

Comment:
note, we need to handle token refreshes. the access token will be fairly short lived. with google it has an auto refresh mechanism. i don't know if there's a similar lib for microsoft that does the same thing